### PR TITLE
AST-628 - fix: WP 5.8 JS console error in customizer due to widget settings.

### PR DESCRIPTION
**Pull request by sagarb90**
### Description
- https://wp-astra.atlassian.net/browse/AST-628
- Customizer not working due to JS error in the wp 5.8 version.
- To fix this issue we have added a separate section for widget design options and a widget block editor.
- Added separate buttons in widget component to navigate to these sections - https://share.getcloudapp.com/eDuyreR1
### Types of changes
 Bug fix (non-breaking change which fixes an issue) 

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/extras.php [hunk 1]
```diff
@@ -647,3 +647,18 @@ function astra_get_transparent_header_default_value() {
...
+
+/**
+ * Check is WordPress version is greater than or equal to beta 5.8 version. 
+ *
+ * @since x.x.x 
+ * @return boolean
+ */
+function astra_is_wp5_8_version() {
+	if ( ( defined( 'GUTENBERG_VERSION' ) && version_compare( GUTENBERG_VERSION, '10.6.2', '>' ) ) 
+	|| version_compare( substr( get_bloginfo( 'version' ), 0, 3 ), '5.8', '>=' ) ) {
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply inc/extras.php:659]: We can use here preg_replace for now, after WP-5.8's some releases we can remove this. Ref @sagarb90 - https://github.com/brainstormforce/astra-addon/blob/master/addons/nav-menu/classes/class-astra-walker-nav-menu-edit-custom.php#L55

File: inc/extras.php [hunk 2]
```diff
@@ -647,3 +647,18 @@ function astra_get_transparent_header_default_value() {
...
 	return apply_filters( 'astra_transparent_header_default_border', $astra_settings['transparent-header-default-border'] );
 }
+
+/**
+ * Check is WordPress version is greater than or equal to beta 5.8 version. 
+ *
+ * @since x.x.x 
+ * @return boolean
+ */
+function astra_is_wp5_8_version() {
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply inc/extras.php:657]: instead of `is` maybe `has` is more correct?I might be wrong

[the remark this answers is not in the document]
Comment (premanshup) [reply inc/extras.php:657]: @Nikschavan I agree! `has` is more accurate.

File: inc/extras.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nikschavan) [reply inc/extras.php:659]: AFAIK `version_compare` also takes care of rc / beta versions into account - eg - https://3v4l.org/n7J5G

Do cross check and test how the version strings of different WP releases is and if that does work with `version_compare`

[the remark this answers is not in the document]
Comment (sagarb90) [reply inc/extras.php:659]: Yes will check the same if version_compare take care for beta and RC versions as well. 
FYI- kadense also used the same method( i.e `preg_replace` ) to compare the beta version. We will change the condition once WP 5.8 released.

[the remark this answers is not in the document]
Comment (sagarb90) [reply inc/extras.php]: I have verified the the version comparision [function ](https://3v4l.org/n7J5G)i.e `version_compare( get_bloginfo( 'version' ), '5.8-alpha', '>=' )` and it works for beta, RC  and stable version numbers. Also updated the same.

File: inc/extras.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (sagarb90) [reply inc/extras.php]: Yes updated to `astra_has_widgets_block_editor`.

**Review by abhishek-bsf — APPROVED** [review]

**Review by imnavanath — APPROVED** [review]

**Pull request merged**